Factor out value refinement to common util

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #57044 Refactor and add handling of boolean comparisons to integer value refinement
* #56966 [JIT] Disable Complete Shape Inlining For Testing Purposes
* #56828 Add unary/binary ops necessary for mobilenet
* #56815 Add Symbolic Shape Registry and define in C++
* #56642 Only optimize while change has been made
* #56438 [JIT] integer value refinement
* #56437 Factor out isDominatedBy
* **#56436 Factor out value refinement to common util**
* #55978 Add Value * == Value * peephole
* #55977 [JIT] [Easy] Factor out non tensor peephole
* #55926 Add list len refinement
* #55925 Add handling of symbolic shapes
* #54809 Initial Symbolic Shape Analysis

Differential Revision: [D27924238](https://our.internmc.facebook.com/intern/diff/D27924238)

Commit:
Factor out value refinement to common util

[ghstack-poisoned]

Commit:
Update on "Factor out value refinement to common util"




[ghstack-poisoned]

Commit:
Update on "Factor out value refinement to common util"




[ghstack-poisoned]

Commit:
Update on "Factor out value refinement to common util"




[ghstack-poisoned]

Commit:
Update on "Factor out value refinement to common util"




[ghstack-poisoned]

Commit:
Update on "Factor out value refinement to common util"




[ghstack-poisoned]

Commit:
Update on "Factor out value refinement to common util"


Differential Revision: [D27924238](https://our.internmc.facebook.com/intern/diff/D27924238)

[ghstack-poisoned]

penguinwu reviewed (approved):
The refactoring and examples make the codes a lot more readable.

Commit:
Update on "Factor out value refinement to common util"


Differential Revision: [D27924238](https://our.internmc.facebook.com/intern/diff/D27924238)

[ghstack-poisoned]
